COMDOX-1722: Ai docs section (#933)
commerce-docs/microsite-commerce-storefront: src/content/docs/ai/build-with-ai.mdx

---
title: Use docs in AI tools
description: Published llms.txt and topic bundles help AI editors and agents use current Adobe Commerce Storefront documentation.
tableOfContents:
  maxHeadingLevel: 2
---

import TableWrapper from '@components/TableWrapper.astro';
import Link from '@components/Link.astro';
import { Aside, Tabs, TabItem } from '@astrojs/starlight/components';

Use this page to connect an AI editor or coding agent to current Adobe Commerce Storefront documentation. With this context, the tool can answer questions about storefront APIs, drop-in versions, and Edge Delivery Services without relying only on general model training.

## What you get

> Use docs context when you want your AI tool to answer from the same storefront documentation you are reading. It helps you:

Reviewer: ```suggestion
Use docs context when you want your AI tool to generate answers based on the storefront documentation you are reading. It helps you:
```

> - Get answers that reflect current storefront docs instead of older model training.

Reviewer: ```suggestion
- Get answers that reflect current storefront documentation instead of older model training.
```

- Point different AI tools at one shared documentation source.
- Scope a chat to a full documentation set or a smaller topic bundle.

## Context files

> Context files are plain-text exports of the pages in this documentation site that an AI tool can read as background information. This site generates them at build time. Start with `llms.txt`, the index that links to every bundle. Full exports contain the whole documentation set. Files under `_llms-txt/` cover one topic area each, so use them when a tool should not load the entire site.

Reviewer: Not sure if you use `-` or `*` for bullets.  What is a bundle?

```suggestion
Context files are plain-text exports of this documentation, regenerated each time the site publishes so AI tools always have the latest content.

- Start with the `llms.txt `file, which serves as the index linking to all bundles.
- Full exports include the entire documentation set.
- Files in `_llms-txt/ `cover individual topic areas—use these when you want to limit the scope and avoid loading the full site.
```
```

<TableWrapper nowrap={[0]}>

| File | Purpose |
|------|---------|

> | [`llms.txt`](/llms.txt) | Index file with an overview and links to every documentation set. |

Reviewer: In line #24, you defined this as `index linking to all bundles` (what is a bundle)  -- what is a documentation set?

| [`llms-full.txt`](/llms-full.txt) | Complete documentation in one file. |
| [`llms-small.txt`](/llms-small.txt) | Abridged bundle (same content as the full export except the long-form release changelog page). |
| `_llms-txt/*.txt` | Topic exports, such as [drop-ins reference](/_llms-txt/dropins-reference.txt), [tutorials](/_llms-txt/tutorials-reference.txt), and [blocks](/_llms-txt/blocks-reference.txt). [`llms.txt`](/llms.txt) lists every bundle web address. |

</TableWrapper>

> Some of these files are large. Loading a whole file sends a lot of text into the model at once. Use context files as a fallback when your AI editor or agent cannot search or fetch the live docs. When possible, browse or search this storefront documentation to see the latest edits, navigation, and diagrams.

Reviewer: Not sure what you mean by this:

When possible, browse or search this storefront documentation to see the latest edits, navigation, and diagrams.

How will they do this?  Do you mean check the release notes, check the GitHub source files for changes?

<Aside type="note" title="Publish timing">

> Context files update each time the site publishes. If a page is edited after the last publish, those changes won't appear in context files until the site publishes again.

Reviewer: How can they check to see what version of the `llm.txt` files they have?  How can they know if there is a more current version, or when a new version is published? Is there a way for the AI to check for updates?

Author: Added this explanation to clarify in the docs: The version they use in their projects is always the latest because the URLs in AGENTS.md or CLAUDE.md are always the latest generated llms files.

## What you get

Use docs context when you want your AI tool to generate answers based on the storefront documentation you are reading. It helps you:

- Get answers that reflect current storefront documentation instead of older model training.
- Point different AI tools at one shared documentation source.
- Scope a chat to a full export or a single topic export.

## Context files

Context files are plain-text exports of this documentation, regenerated each time the site publishes so AI tools always have the latest content.

- Start with `llms.txt`, the index that links to every export file.
- Full exports contain the entire documentation site in one file.
- Topic exports under `_llms-txt/` each cover one subject area. Use them when a tool should not load the entire site.

<TableWrapper nowrap={[0]}>

| File | Purpose |
|------|---------|
| [`llms.txt`](/llms.txt) | Index with an overview and links to every export file. |
| [`llms-full.txt`](/llms-full.txt) | Complete documentation in one file. |

> | [`llms-small.txt`](/llms-small.txt) | Same as the full export but without the release changelog, which is very long. Use this when a tool has a context size limit. |

Reviewer: ```suggestion
| [`llms-small.txt`](/llms-small.txt) | Same as the full export but without the release changelog, which is very long. Use this when your AI  tool has a context size limit. |
```

```markdown wrap
## Adobe Commerce Storefront Documentation

See https://experienceleague.adobe.com/developer/commerce/storefront/llms.txt for Adobe Commerce Storefront documentation.
```

Cursor agent mode reads `AGENTS.md` and fetches the index when needed. You can also add this reference to a file under `.cursor/rules/` if you prefer a Cursor-specific location.

</TabItem>
<TabItem label="Claude Code">

Create or update `CLAUDE.md` at your project root and add a line pointing to the documentation index:

```markdown wrap
See https://experienceleague.adobe.com/developer/commerce/storefront/llms.txt for Adobe Commerce Storefront documentation.
```

Claude Code fetches the index and linked bundles when the documentation is relevant. To confirm the setup, start a new chat and ask a question this storefront documentation answers, such as a question about a drop-in. If the reply cites or matches those topics, the tool can read the documentation.

</TabItem>
<TabItem label="Other AI tools">

Other AI editors and agents often accept a documentation URL in settings or system prompts.

> For tools that support custom context URLs or documentation sources, such as GitHub Copilot Chat or Windsurf, provide this URL below. Tools that implement the <Link href="https://llmstxt.org" text="llms.txt spec" /> can follow the index to the relevant bundles automatically.

Reviewer: ```suggestion
Other AI editors and agents often accept a documentation URL in settings or system prompts. Tools that implement the <Link href="https://llmstxt.org" text="llms.txt spec" /> can follow the index to the relevant bundles automatically.

For tools that support custom context URLs or documentation sources, such as GitHub Copilot Chat or Windsurf, provide the following URL:
```

```text
https://experienceleague.adobe.com/developer/commerce/storefront/llms.txt
```

For tools that do not support URL-based context, paste the content of `llms-small.txt` or a relevant `_llms-txt/*.txt` topic bundle directly into the chat or system prompt.

</TabItem>
</Tabs>

## Example prompt

> After you add the documentation context, test the connection with a concrete question in chat. You can paste or rewrite the sample below.

Reviewer: Not sure how well "concrete question" will translate.

```suggestion
After you add the documentation context, validate it by asking a specific question in chat. You can use or adapt the example below.
```

```bash wrap
Use the Storefront docs. I'm building a product detail page and I want to add a custom 'Add to Wishlist' button in the product details drop-in, below the existing Add to Cart button. How do I use slots to do this?
```

## Check the result

> A successful setup produces replies that cite storefront topics, drop-in APIs, or paths that match this documentation. Adapt drop-in names or UI labels if your project differs. Then verify the reply against the live pages.

Reviewer: ```suggestion
A successful setup returns responses that reference storefront topics, drop-in APIs, or paths aligned with this documentation. Adapt drop-in names or UI labels as needed for your project, then validate the response against the live documentation.
```
